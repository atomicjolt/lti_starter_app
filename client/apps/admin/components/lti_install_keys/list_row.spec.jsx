import React from 'react';
import TestRenderer from 'react-test-renderer';
import ListRow from './list_row';

describe('application instances list row', () => {
  let props;
  let result;
  let instance;
  let deleted;

  beforeEach(() => {
    deleted = false;
    props = {
      delete: () => { deleted = true; },
      save: () => {},
      ltiInstallKey: {
        id: 2,
        application_id: 23,
        clientId: 'lti-key',
        iss: 'iss',
        jwksUrl: 'jwksUrl',
        tokenUrl: 'tokenUrl',
        oidcUrl: 'oidcUrl',
        created_at: 'created_at',
      },
    };
    result = TestRenderer.create(<ListRow {...props} />);
    instance = result.root;
  });

  // the following tests will break if the order of the buttons is changed
  // to remedy this a class would need to be added to each button

  it('handles the opening of the modal', () => {
    expect(instance.state.modalOpen).toBeFalsy();
    const btn = instance.findByType('button').first();
    btn.props.onClick();
    expect(result.root.state.modalOpen).toBeTruthy();
  });

  it('handles deleting', () => {
    expect(deleted).toBeFalsy();
    expect(instance.state.confirmDeleteModalOpen).toBeFalsy();
    const btn = instance.findByType('button').last();
    btn.props.onClick();
    expect(result.root.state.confirmDeleteModalOpen).toBeTruthy();
  });
});
