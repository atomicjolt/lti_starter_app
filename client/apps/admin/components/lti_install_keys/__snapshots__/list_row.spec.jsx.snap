// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application instances list row handles the opening of the modal 1`] = `
<tr>
  <td>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "grey",
          "cursor": "pointer",
          "fontSize": "1.5em",
        }
      }
      type="button"
    >
      <i
        className="material-icons"
      >
        settings
      </i>
    </button>
  </td>
  <td>
    2
  </td>
  <td>
    iss
  </td>
  <td>
    lti-key
  </td>
  <td>
    Invalid Date
     
    Invalid Date
  </td>
  <td>
    <button
      className="c-delete"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "grey",
          "cursor": "pointer",
          "fontSize": "1.5em",
        }
      }
      type="button"
    >
      <i
        className="i-delete"
      />
    </button>
  </td>
</tr>
`;

exports[`application instances list row handles the opening of the modal 2`] = `
<tr>
  <td>
    <button
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "grey",
          "cursor": "pointer",
          "fontSize": "1.5em",
        }
      }
      type="button"
    >
      <i
        className="material-icons"
      >
        settings
      </i>
    </button>
    <div
      className="ReactModal__Overlay c-modal__background"
      onClick={[Function]}
      onMouseDown={[Function]}
      style={Object {}}
    >
      <div
        aria-label="Lti Install Key Modal"
        aria-modal={true}
        className="ReactModal__Content c-modal c-modal--settings is-open"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        role="dialog"
        style={Object {}}
        tabIndex="-1"
      >
        <h2
          className="c-modal__title"
        >
          Update
           
          Lti Install Key
        </h2>
        <form>
          <div
            className="o-grid o-grid__modal-top"
          >
            <div
              className="o-grid__item u-half"
            >
              <div
                className="c-input--container"
              >
                <label
                  className="c-input"
                  htmlFor="iss_input"
                >
                  <span>
                    iss
                  </span>
                  <input
                    id="iss_input"
                    name="iss"
                    onChange={[Function]}
                    type="text"
                    value="iss"
                  />
                </label>
              </div>
            </div>
            <div
              className="o-grid__item u-half"
            >
              <div
                className="c-input--container"
              >
                <label
                  className="c-input"
                  htmlFor="client_id_input"
                >
                  <span>
                    client id
                  </span>
                  <input
                    id="client_id_input"
                    name="clientId"
                    onChange={[Function]}
                    type="text"
                    value="lti-key"
                  />
                </label>
              </div>
            </div>
            <div
              className="o-grid__item u-half"
            >
              <div
                className="c-input--container"
              >
                <label
                  className="c-input"
                  htmlFor="jwks_url_input"
                >
                  <span>
                    jwks url
                  </span>
                  <input
                    id="jwks_url_input"
                    name="jwksUrl"
                    onChange={[Function]}
                    type="text"
                    value="jwksUrl"
                  />
                </label>
              </div>
            </div>
            <div
              className="o-grid__item u-half"
            >
              <div
                className="c-input--container"
              >
                <label
                  className="c-input"
                  htmlFor="token_url_input"
                >
                  <span>
                    token url
                  </span>
                  <input
                    id="token_url_input"
                    name="tokenUrl"
                    onChange={[Function]}
                    type="text"
                    value="tokenUrl"
                  />
                </label>
              </div>
            </div>
            <div
              className="o-grid__item u-half"
            >
              <div
                className="c-input--container"
              >
                <label
                  className="c-input"
                  htmlFor="oidc_url_input"
                >
                  <span>
                    oidc url
                  </span>
                  <input
                    id="oidc_url_input"
                    name="oidcUrl"
                    onChange={[Function]}
                    type="text"
                    value="oidcUrl"
                  />
                </label>
              </div>
            </div>
          </div>
          <button
            className="c-btn c-btn--yellow"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
          <button
            className="c-btn c-btn--gray--large u-m-right"
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </form>
      </div>
    </div>
  </td>
  <td>
    2
  </td>
  <td>
    iss
  </td>
  <td>
    lti-key
  </td>
  <td>
    Invalid Date
     
    Invalid Date
  </td>
  <td>
    <button
      className="c-delete"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": "none",
          "color": "grey",
          "cursor": "pointer",
          "fontSize": "1.5em",
        }
      }
      type="button"
    >
      <i
        className="i-delete"
      />
    </button>
  </td>
</tr>
`;
